atlp-rwanda/ecommerce-app-predators-bn
# [24] [Ch setup testing environment]

#### What does this PR do?
  This PR adds a testing environment to ensure all functionalities are tested before deployment.

#### Description of Task to be completed?
Installed and configured the testing environment using [Mocha, Chai, Sinon, Jest and Supertest].
Added tests for all functionalities to ensure they are working correctly.

#### How should this be manually tested?
Checkout this branch.
Run npm install to install all dependencies.
Run npm test to run all tests.
Check the console output for test results.

#### Any background context you want to provide?

Testing is an important part of the development process, as it helps ensure that all functionalities are working as expected and reduces the likelihood of bugs being introduced during deployment. By adding a testing environment, we can automate the testing process and catch any issues before they become a problem.
#### What are the relevant pivotal tracker/Trello stories?
N/A
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A
